File: src/Level1/정수제곱근판별.java
package Level1;

public class 정수제곱근판별 {
    public long solution(long n) {
        if(Math.pow((long)Math.sqrt(n), 2) == n)
            return (long)Math.pow(Math.sqrt(n) + 1, 2);
        return -1;
    }
}

File: src/Level1/정수제곱근판별.kt
fun main(){
    println(solution(100))
    println(solution(3))
    println(solution(4))
}

private fun solution(n: Long): Long{
    val root = Math.sqrt(n.toDouble()).toLong()
    return if(n == root.times(root))
        (root.inc()).times(root.inc())
    else -1
}